### DISCUSSION TITLE
Best practice to manage bank transfer

### DISCUSSION THREAD

**Discussion by fedex03** [firefly-iii/firefly-iii#7105] [Firefly III]
Hi there,

I have two bank account Bank1 and Bank2.
Every month I move, via Bank Transfer, some money from Bank1 to Bank2.

This involvs:

- Bank1: a withdrawal transaction 
- Bank2: a deposit transaction.

I tried to usa the function "Trasnfer" on a rule on Bank1.

This works on Bank1 but on Bank2 I have two transaction:

- A transfer
- A deposit.

This situation doubles my account money.

I tought to add a rule on Bank2 to remove this specific deposit.
But I don't know if this is the correct approach,

Thank you for your help!

**Comment by JC5** [comment]
That would be the correct approach, yes.

Are you importing CSV files?

**Reply by fedex03** [reply]
Thank you!
Yes, I import CSV file with the Data Importer.

**Comment by marcoil** [comment]
I have a similar case, from a bank account to PayPal, but I use Spectre for both. I use rules to convert the bank withdrawal to a transfer and then delete the paypal deposit. Everything works but then the importer complains and sends me an email message with "Cannot find transaction. Possibly, a rule deleted this transaction after its creation."

Am I doing something wrong?

**Reply by JC5** [reply]
If the deposit is deleted by a rule directly after it's created, this will happen.

**Reply by marcoil** [reply]
Is there a way to avoid those messages? Or any other way of not having those transactions appear as withdrawal/deposits in the financial report?

**Reply by JC5** [reply]
If you have enough information to save the transaction as a transfer right from the start then yes. Otherwise, sorry.

**Reply by marcoil** [reply]
> If you have enough information to save the transaction as a transfer right from the start then yes.

Yes, I have all the information to make a transfer from one of the transactions. I just don't know how to handle the other transaction (the one that goes the other way) in a way that doesn't send an email message for each one.

Any pointers would be greatly appreciated.

**Reply by marcoil** [reply]
I think I found a solution to my problem. In order to avoid errors due to the second transaction being deleted at creation, I tried this and it seems to work:

- Use a creation rule to convert one of the transactions into a transfer.
- Add another creation to rule to mark the other transaction with a category or tag that's only used for these cases. In my case I use a category *To delete*.
- In a different rule **group**, add a rule for updated transactions that detects that category or tag and deletes the transaction.

That seems to get rid of the redundant transaction and the warning messages.
